src/basilisp/core.lpy

```diff
@@ -1606,7 +1606,7 @@
   [x]
   (python/int x))
 
-(defn ^:inline boolean
+(defn ^:no-inline boolean
```

Comment:
When this function is inlined with the constants `0` and `0.0`, the generated code for calls to `false?` becomes `False == 0` and `False == 0.0` respectively with the changes in this PR. In Python, those expressions are always true. The way we distinguish between them for Clojure compatibility (since only `nil` and `false` are logical false in Clojure) is to use `is`, but when using inlined constants that generates a syntax warning.

The easiest fix is to just not inline the function call since using `is` on a variable (even one containing `0` or `0.0`) is legal syntax.